lib/middleware/request_tracker.rb

```diff
@@ -61,7 +61,7 @@ def self.log_request(data)
       elsif data[:has_auth_cookie]
         ApplicationRequest.increment!(:page_view_logged_in)
         ApplicationRequest.increment!(:page_view_logged_in_mobile) if data[:is_mobile]
-      else
+      elsif !(GlobalSetting.try(:ignore_anonymous_pageviews) && SiteSetting.login_required)
```

Comment:
no need for a `try` here, once it exists in discourse_defaults it is safe to use.

That said, I say we just kill the setting altogether, it is not named right. 

`ignore_anon_views_on_private_sites` is probably more correct, but it a mouthful. 

Let's offer no setting, then if someone complains they need a setting we can consider adding it.